src/platform/site-wide/announcements/components/VAPlusVetsModal.jsx

```diff
@@ -10,7 +10,7 @@ export default class VAPlusVetsModal extends React.Component {
     let wasRedirectedFromVets = !!referrer && referrer.includes('vets.gov');
 
     // Allow an override on the URL to force the Onboarding Modal to appear for testing purposes.s
-    if (__BUILDTYPE__ !== 'preview') {
+    if (__BUILDTYPE__ !== 'preview' && !wasRedirectedFromVets) {
       wasRedirectedFromVets = window.location.search.includes(
         'onboarding-modal',
       );
```

Comment:
I discovered that we should only check the test-only flag if we've confirmed that they weren't redirected via the referrer already. Ultimately, we can use this PR's review instance to verify things are working by visiting `https://vetsgov-pr-8948.herokuapp.com/va-gov-onboarding-modal-check?vets.gov`.